### PR TITLE
[db] Eliminating duplicate key constraint violations

### PR DESCRIPTION
**Pull request by bruntib**
In concurrent storage of two runs containing the same files leads to
duplicate key constraint violation. Some DBMSs can gracefully handle
this issue by supporting "ON CONFLICT DO NOTHING" clause at INSERT
statement.

### CODE REVIEW

**Review by vodorok — APPROVED** [review]
LGTM

File: web/server/codechecker_server/api/mass_store_run.py [hunk 1]
```diff
@@ -152,8 +152,26 @@ def add_file_record(
...
+        # supporting "ON CONFLICT DO NOTHING" clause in INSERT statement which
+        # solves the same issue gracefully.
+        # TODO: "ON CONFLICT DO NOTHING" feature is available for SQLite engine
+        # too in SQLAlchemy 1.4.
+        if session.bind.dialect.name == 'postgresql':
+            insert_stmt = sqlalchemy.dialects.postgresql.insert(File).values(
+                filepath=file_path,
+                filename=os.path.basename(file_path),
+                content_hash=content_hash).on_conflict_do_nothing(
+                    index_elements=['id'])
```
Comment (dkrupp) [inline web/server/codechecker_server/api/mass_store_run.py:168]: index_elements=["content_hash"]?? Shouldn't we list the key clashing constraint fields here?

**Review by dkrupp — CHANGES_REQUESTED** [review]
Please add test cases. Please reproduce the error and check if it is solved after the change.

File: web/server/codechecker_server/api/mass_store_run.py (hunk 1 above)
Comment (dkrupp) [reply web/server/codechecker_server/api/mass_store_run.py:168]: Maybe you can test this fix and provoke the error by storing a file to the server (with masstore run) that has been stored already.

Comment (bruntib) [reply web/server/codechecker_server/api/mass_store_run.py:168]: [2022-08-15 09:30:01 UTC] <???> (23505) - ERROR:  duplicate key value violates unique constraint "uq_files_filepath"
[2022-08-15 09:30:01 UTC] <???> (23505) - DETAIL:  Key (filepath, content_hash)=(/path/to/blabla.h, <hash_value>) already exists.
[2022-08-15 09:30:01 UTC] <???> (23505) - STATEMENT:  INSERT INTO files (filepath, filename, content_hash, remote_url, tracking_branch) VALUES ('/path/to/blabla.h', 'blabla.h', '<hash_value>', NULL, NULL) RETURNING files.id

**Pull request merged**